shared_model/interfaces/iroha_internal/block.hpp

```diff
@@ -15,15 +15,70 @@
  * limitations under the License.
  */
 
-#ifndef IROHA_BLOCK_HPP
-#define IROHA_BLOCK_HPP
+#ifndef IROHA_SHARED_MODEL_BLOCK_HPP
+#define IROHA_SHARED_MODEL_BLOCK_HPP
+
+#include "interfaces/signable.hpp"
+#include "interfaces/transaction.hpp"
+
 namespace shared_model {
   namespace interface {
 
-    class Block {
-      // TODO implement
+    class Block : public Signable<Block, iroha::model::Block> {
+     public:
+      /// Block number (height) type
+      using BlockHeightType = uint64_t;
+
+      /**
+       * @return block number in the ledger
+       */
+      virtual BlockHeightType &height() const = 0;
+
+      /**
+       * @return hash of a previous block
+       */
+      virtual HashType &prevHash() const = 0;
+
+      /// Type of a number of transactions in block
+      using TransactionsNumberType = uint16_t;
+
+      /**
+       * @return amount of transactions in block
+       */
+      virtual TransactionsNumberType &txsNumber() const = 0;
+
+      /**
+       * @return Root of merkle tree based on the block and all previous blocks
+       * in the ledger
+       */
+      virtual HashType &merkleRoot() const = 0;
+
+      /// Type of transactions' collection
+      using TransactionsType = std::vector<Transaction>;
+
+      /**
+       * @return collection of transactions
+       */
+      virtual TransactionsType &transactions() const = 0;
+
+      iroha::model::Block *makeOldModel() const {
+        iroha::model::Block *oldStyleBlock;
```

Comment:
Is there the allocation of `oldStyleBlock`?

Reply:
Allocation on heap assumed. This is an error.

Reply:
Sure. Already fixed it locally.